pandas/core/arrays/arrow/array.py

```diff
@@ -2579,7 +2579,7 @@ def _str_split(
             n = None
         if pat is None:
             split_func = pc.utf8_split_whitespace
-        elif regex:
+        elif regex or (regex is None and len(pat) != 1):
```

Comment:
Thanks for the PR - I'm not sure this is the right fix though. Do you see where the behavior deviates between the different string types? This current fix seems like it would apply a behavior change to all types

Reply:
Thanks for the review. The behavior deviates [here](https://github.com/pandas-dev/pandas/blob/a1fc8e8147efb0c7d7e10e674c3ee383b14f2d43/pandas/core/strings/accessor.py#L913).

`string[pyarrow]` goes through https://github.com/pandas-dev/pandas/blob/a1fc8e8147efb0c7d7e10e674c3ee383b14f2d43/pandas/core/strings/object_array.py#L327
while `pd.ArrowDtype(pa.string())` goes through
https://github.com/pandas-dev/pandas/blob/a1fc8e8147efb0c7d7e10e674c3ee383b14f2d43/pandas/core/arrays/arrow/array.py#L2571

The docstring of `str.split` says this about `regex`: "If None and pat length is not 1, treats pat as a regular expression."

This behavior has been implemented in the first `_str_split`, but not in the second `_str_split`. So I add this condition in the second `_str_split` to fix the issue.

Reply:
Ah OK thanks that is helpful. Is there a way to make these implementations look more alike? I see what you are trying to accomplish here but its hard to tell the corner cases where these may still diverge. Is there a reason why the implementations need to differ at all?

Reply:
My initial intention was to make as few changes as possible.

To make it more coherent, I would rather set `regex=True` for the corner case before calling `_str_split` in the code below. Do you think it's OK?
https://github.com/pandas-dev/pandas/blob/a1fc8e8147efb0c7d7e10e674c3ee383b14f2d43/pandas/core/strings/accessor.py#L911-L913

Reply:
I move outside the logit that determines if `pat` is a regex, so that the two `_str_split` look more alike. Coud you review again?